.github/workflows/rust.yml

```diff
@@ -42,11 +42,16 @@ defaults:
     working-directory: rust
 
 jobs:
+  native-unix:
+    uses: ./.github/workflows/native-unix.yml
+
   rust:
+    needs: [native-unix]
     strategy:
       matrix:
-        # TODO(lidavidm): add back macos-latest once it has Cargo installed
-        os: [windows-latest, macos-13, ubuntu-latest]
+        # TODO(alexandreyc): add `windows-latest`
+        # See: https://github.com/apache/arrow-adbc/pull/1803#issuecomment-2117669300
+        os: [macos-latest, ubuntu-latest]
```

Comment:
Keep macos-13, or else we aren't testing on x64 macos?

Reply:
Done!